File: EDR_NDR/frontend/src/types/agent.ts
export type AgentStatus = 'online' | 'offline' | 'isolated'

export interface AgentRef {
  id: string
  hostname: string
  ip_address: string
  status: AgentStatus
}

export interface Agent {
  id: string
  hostname: string
  ip_address: string
  os: string
  os_version: string
  status: AgentStatus
  agent_version: string
  last_seen: string
  group_tag: string
  risk_score: number
}

export interface AgentDetail extends Agent {
  open_alert_count: number
  total_alert_count: number
}

export interface AgentListParams {
  q?: string
  status?: AgentStatus
  page?: number
  page_size?: number
}

File: EDR_NDR/backend/app/schemas/agent.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import AgentStatus


class AgentRef(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    hostname: str
    ip_address: str
    status: AgentStatus


class Agent(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    hostname: str
    ip_address: str
    os: str
    os_version: str
    status: AgentStatus
    agent_version: str
    last_seen: datetime
    group_tag: str
    risk_score: int


class AgentDetail(Agent):
    open_alert_count: int
    total_alert_count: int
